### PR TITLE
Add Holocene fork references

### PR DESCRIPTION
**Pull request by sebastianst**
**Description**

* Updates SCR to pull in new Holocene fork field.
* Adds Holocene fork references in various places.

**Additional context**

Part of Holocene prep

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -1952,7 +1957,7 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 			if rawdb.ReadCanonicalHash(chaindb, 0) != (common.Hash{}) {
 				cfg.Genesis = nil // fallback to db content
 
-				//validate genesis has PoS enabled in block 0
+				// validate genesis has PoS enabled in block 0
```
Comment (axelKingsley) [inline cmd/utils/flags.go:2025]: 😎

File: go.mod [hunk 2]
```diff
@@ -22,7 +22,7 @@ require (
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc
 	github.com/deckarep/golang-set/v2 v2.1.0
 	github.com/dop251/goja v0.0.0-20230806174421-c933cf95e127
-	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20240803025447-c92ef420eec2
+	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20240806205917-f72450b810e2
```
Comment (axelKingsley) [inline go.mod]: I assume the SCR reference is changing because it needs to point at a build familiar with Holocene?

File: params/config.go [hunk 3]
```diff
@@ -1090,7 +1105,7 @@ type Rules struct {
 	IsVerkle                                                bool
 	IsOptimismBedrock, IsOptimismRegolith                   bool
 	IsOptimismCanyon, IsOptimismFjord                       bool
-	IsOptimismGranite                                       bool
+	IsOptimismGranite, IsOptimismHolocene                   bool
```
Comment (axelKingsley) [inline params/config.go:1125]: nit: why two to a line?

**Review by axelKingsley — APPROVED** [review]
All changes have strong prior-art so I was able to confirm that the Holocene changes match in all cases with the treatment of other Hard Forks. In particular I confirmed that no Copy/Paste errors misnamed any lines.

All mechanical changes appear correct. I wonder if someday we could automate this.

File: cmd/utils/flags.go (hunk 1 above)
Comment (sebastianst) [reply cmd/utils/flags.go:2025]: that was actually my linter...

File: params/config.go (hunk 3 above)
Comment (sebastianst) [reply params/config.go:1125]: simply because it was done that way for prior forks 😬

File: go.mod (hunk 2 above)
Comment (sebastianst) [reply go.mod]: yep, that's the version that introduced the `HoloceneTime` field. But I may want to update the reference again, and there are other go.mod merge conflicts to resolve...

**Pull request merged**